YoutubeDownloader/Views/Dialogs/AuthSetupView.xaml.cs

```diff
@@ -30,53 +37,60 @@ private void LogoutHyperlink_OnClick(object sender, RoutedEventArgs args)
 
     private void WebBrowser_OnLoaded(object sender, RoutedEventArgs args) => NavigateToLoginPage();
 
-    private void WebBrowser_OnCoreWebView2InitializationCompleted(
-        object? sender,
-        CoreWebView2InitializationCompletedEventArgs args
-    )
+    private void WebBrowser_OnWebViewCreated(object sender, WebViewCreatedEventArgs args)
     {
-        if (!args.IsSuccess)
+        if (!args.IsSucceed)
             return;
 
-        WebBrowser.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
-        WebBrowser.CoreWebView2.Settings.AreDevToolsEnabled = false;
-        WebBrowser.CoreWebView2.Settings.IsGeneralAutofillEnabled = false;
-        WebBrowser.CoreWebView2.Settings.IsPasswordAutosaveEnabled = false;
-        WebBrowser.CoreWebView2.Settings.IsStatusBarEnabled = false;
-        WebBrowser.CoreWebView2.Settings.IsSwipeNavigationEnabled = false;
+        var platformWebView = (
+            (Avalonia.WebView.Windows.Core.WebView2Core)WebBrowser.PlatformWebView!
+        );
+        _coreWebView2 = platformWebView!.CoreWebView2!;
+
+        _coreWebView2.Settings.AreDefaultContextMenusEnabled = false;
+        _coreWebView2.Settings.AreDevToolsEnabled = false;
+        _coreWebView2.Settings.IsGeneralAutofillEnabled = false;
+        _coreWebView2.Settings.IsPasswordAutosaveEnabled = false;
+        _coreWebView2.Settings.IsStatusBarEnabled = false;
+        _coreWebView2.Settings.IsSwipeNavigationEnabled = false;
     }
 
-    private void WebBrowser_OnNavigationStarting(
+    private async void WebBrowser_OnNavigationStarting(
         object? sender,
-        CoreWebView2NavigationStartingEventArgs args
+        WebViewUrlLoadingEventArg args
     )
     {
         // Reset existing browser cookies if the user is attempting to log in (again)
-        if (string.Equals(args.Uri, LoginPageUrl, StringComparison.OrdinalIgnoreCase))
-            WebBrowser.CoreWebView2.CookieManager.DeleteAllCookies();
+        if (string.Equals(args.Url?.AbsoluteUri, LoginPageUrl, StringComparison.OrdinalIgnoreCase))
+            _coreWebView2!.CookieManager.DeleteAllCookies();
+
+        if (args.Url!.AbsoluteUri.StartsWith(HomePageUrl, StringComparison.OrdinalIgnoreCase))
+        {
+            // Extract the cookies that the browser received after logging in
+            var cookies = await _coreWebView2!.CookieManager.GetCookiesAsync(
+                args.Url!.AbsoluteUri!
+            );
+            ViewModel.Cookies = cookies.Select(c => c.ToSystemNetCookie()).ToArray();
+        }
```

Comment:
Does this need to happen in `OnNavigationStarting` as well as `OnNavigationCompleted`?

Reply:
removed OnNavigationCompleted as OnNavigationStarting does the work. OnNavigationCompleted does not reach the codepath where the cookies are set, I have kept it because in WPF version that is the event that is used.